src/ccid_serial.c

```diff
@@ -787,7 +788,8 @@ status_t OpenSerialByName(unsigned int reader_index, char *dev_name)
 	 * will echo characters for you.  Don't generate signals. */
 	current_termios.c_lflag = 0;
 
-	if (0 == strcasecmp(reader_name,"GemCoreSIMPro2"))
+	readerID = serialDevice[reader].ccid.readerID;
+	if (readerID == GEMCORESIMPRO2)
```

Comment:
I prefer to have `if (GEMCORESIMPRO2 == readerID)` to generate a compilation error in case I use `=` instead of `==`.
It is defensive programmation.